lib/coxir/struct/guild.ex

```diff
@@ -34,6 +34,31 @@ defmodule Coxir.Struct.Guild do
     |> replace(:roles, &Role.get/1)
   end
 
+  @doc """
+  Creates a guild with the given information
+
+  Returns a guild object upon success
+  or a map containing error information.
+
+  #### Params
+  Must be an enumerable with any of the fields listed below.
+  - `name` - name of the created guild
+  - `region` - voice region id in the created guild
+  - `icon` - base64 128x128 jpeg image icon in the created guild
+  - `verification_level` - the verification level in the created guild
+  - `default_message_notifications` - the default notification level in the created guild
+  - `explicit_content_filter` - the explicit content filter level in the created guild
+  - `roles` - the roles in the created guild
+  - `channels` - the channels in the created guild
+
+  Refer to [this](https://discordapp.com/developers/docs/resources/guild#create-guild)
+  for a broader explanation on the fields and their defaults.
+  """
+  @spec create(Enum.t) :: map
+  def create(params) do
+    API.request(:POST, "/guilds", params)
```

Comment:
What about following the function style of https://github.com/satom99/coxir/blob/master/lib/coxir/struct/guild.ex#L81 for functions containing one single line of code?

Reply:
I believe we only do this when we are referencing another function. There's a few times this happens below from where I linked.

https://github.com/satom99/coxir/blob/master/lib/coxir/struct/guild.ex#L178

Reply:
If you check this PR you will see we followed the style we mentioned. https://github.com/satom99/coxir/pull/5/files

Reply:
Yes that's what I mean.

When we have a function just act as a wrapper for another function,we use the `do:` syntax.  For example:
```elixir
  def edit(%{id: id}, params),
    do: edit(id, params)

  def edit(guild, params) do
    API.request(:patch, "guilds/#{guild}", params)
    |> pretty
  end
```

That PR there is somewhat similar, but it's pretty borderline and fine either way. The rest of the codebase when having a one line function that isn't a wrapper on-top of another function does not use the `do:` syntax. For example:
```elixir
  def delete(%{id: id}),
    do: delete(id)

  def delete(guild) do
    API.request(:delete, "guilds/#{guild}")
  end

  def create_channel(%{id: id}, params),
    do: create_channel(id, params)

  def create_channel(guild, params) do
    API.request(:post, "guilds/#{guild}/channels", params)
  end

  def bulk_delete_messages(%{id: id}, messages),
    do: bulk_delete_messages(id, messages)

  def bulk_delete_messages(channel, messages) do
    API.request(:post, "channels/#{channel}/messages/bulk-delete", %{messages: messages})
  end

  def typing(%{id: id}),
    do: typing(id)

  def typing(channel) do
    API.request(:post, "channels/#{channel}/typing")
  end

  # and so on 
```